### DISCUSSION TITLE
can I set different thresholds for x and y with Plot.bin?

### DISCUSSION THREAD

**Discussion by enjalot** [observablehq/plot#872] [Q&A] [answered]
If I'm binning across x and y, can i set the thresholds individually?

for example:
```Plot.rect(sales, Plot.bin({fill: "count"},{x: "date", y: "medianAskingPrice", thresholds: 30}))```

**Comment by mbostock** [comment, marked as answer]
Yes, e.g.

```js
Plot.rect(sales, Plot.bin({fill: "count"},{x: {value: "date", thresholds: 30}, y: "medianAskingPrice"}))
```

From the [README](https://github.com/observablehq/plot/blob/main/README.md#bin):

> To pass separate binning options for x and y, the x and y input channels can be specified as an object with the options above and a value option to specify the input channel values.
>
> ```js
> Plot.binX({y: "count"}, {x: {thresholds: 20, value: "culmen_length_mm"}})
> ```

**Reply by enjalot** [reply]
ah thank you! i was reading closely *under* the thresholds options 🤦‍♂️
